chia/wallet/wallet_node.py

```diff
@@ -1003,13 +1003,12 @@ async def get_timestamp_for_height_from_peer(self, height: uint32, peer: WSChiaC
         a transaction block
         """
         cache = self.get_cache_for_peer(peer)
-        cached_timestamp = cache.get_height_timestamp(height)
-        if cached_timestamp is not None:
-            return cached_timestamp
-
         last_tx_block = None
         request_height: int = height
         while request_height >= 0:
+            cached_timestamp = cache.get_height_timestamp(uint32(request_height))
+            if cached_timestamp is not None:
+                return cached_timestamp
```

Comment:
I'm not sure this will have the desired effect. I believe this will only give you a timestamp if `request_height` refers to a tx-block, right?

Which means you'll still keep requesting blocks from the peer until you find the tx-block. You'll perform the same search next time, because the non-tx-blocks still won't have timestamps

Reply:
Yeah right! I added 0aa3dea76a3a733f30d50eee500278b763d915d6 to make use of the cached blocks.